app/services/force/listing_service.rb

```diff
@@ -29,36 +31,70 @@ def self.eligible_listings(filters)
     # get one detailed listing result by id
     def self.listing(id, opts = {})
       @cache = Rails.cache
-      endpoint = "/ListingDetails/#{CGI.escape(id)}"
+      endpoint = "#{CACHE_KEY_PREFIX}#{CGI.escape(id)}"
       force = opts[:force] || false
       Rails.logger.info("Calling self.listing for #{id} with force: #{force}")
+
       results = Request.new(parse_response: true).cached_get(endpoint, nil, force)
+      listing = process_listing_images(results)
+
+      listing['translations'] = get_listing_translations(listing) || {}
+      listing
+    end
+
+    def self.process_listing_images(results)
       results_with_cached_listing_images = add_cloudfront_urls_for_listing_images(results)
-      listing = add_image_urls(results_with_cached_listing_images).first
+      add_image_urls(results_with_cached_listing_images).first
+    end
 
-      if ::UNLEASH.is_enabled? 'GoogleCloudTranslate'
-        listing_translations = @cache.fetch("/ListingDetails/#{id}/translations") do
-          Rails.logger.info("Nothing in cache for Listing #{id} translations")
-          {}
-        end
+    def self.get_listing_translations(listing)
+      return unless ::UNLEASH.is_enabled? 'GoogleCloudTranslate'
 
-        unless translations_valid?(listing_translations)
-          Rails.logger.info("Translations are not valid for #{id}")
-          results = Request.new(parse_response: true).cached_get(endpoint, nil, true)
-          listing_translations = CacheService.new.process_translations(results.first)
-        end
+      listing_id = listing['Id']
+      listing_translations = fetch_listing_translations_from_cache(listing_id)
+      if translations_invalid?(listing_translations) || translations_are_outdated?(
+        listing_translations[:LastModifiedDate], listing['LastModifiedDate']
+      )
+        Rails.logger.info("Translations are not valid for #{listing_id}")
+        return CacheService.new.process_translations(listing)
+      end
 
-        listing['translations'] = listing_translations || {}
+      if listing_is_outdated?(listing_translations[:LastModifiedDate],
+                              listing['LastModifiedDate'])
+        Rails.logger.info("Listing is outdated for #{listing_id}")
+        refresh_listing_cache(listing_id)
```

Comment:
I thought we discussed not refreshing the listing cache in the translations flow?

Reply:
We removed it from the translation service. We do want to refresh the listing when we know that the listing is outdated.

Reply:
That wasn't my understanding. I thought we wanted to avoid updating the listing cache on page load - if we update on page load than the cache is moot.

Reply:
I discussed this above: 

> Maybe we should make the LastModifiedDate comparison have two outcomes: if the listing date is older than the translations -> refresh the listing. If the translation date is older than the listing -> refresh the translations.

But it is a fair question. Relates to this ticket as well: https://sfgovdt.jira.com/browse/DAH-2575

We can wait for the 10 minutes: which means english values will be out of date until a listing recache if we update translations via an event. If we have an event miss, then we would have to wait the 10 minutes to resolve those differences as well.

Reply:
We discussed in a meeting and decided to move forward with updating the listing cache!